CMaxTable.lua

```diff
@@ -19,8 +19,7 @@ end
 
 function CMaxTable:updateGradInput(input, gradOutput)
    for i=1,#input do
-      self.gradInput[i] = torch.Tensor()
-      self.gradInput[i]:resizeAs(input[i]):fill(0.0)
+      self.gradInput[i] = input[i]:clone():fill(0.0)
       local mask = torch.eq(self.maxIdx, i)
```

Comment:
yes, having a buffer called `self.mask` for example, and reusing that would be enough. But there is a new memory allocator in the works, so that's why I was asking @soumith if he thinks it's necessary to do this change.